### PR TITLE
RegexArrayShapeMatcher - Support 'n' modifier

### PR DESCRIPTION
**Pull request by staabm**
review commit by commit

### CODE REVIEW

**Comment by staabm** [comment]
@Seldaek with this PR we can now take regex pattern modifiers into account.
this should help with implementing https://github.com/phpstan/phpstan/issues/11360

**Comment by ondrejmirtes** [comment]
Thank you!

Please next time instead of separate commits, send separate PRs.

File: src/Type/Php/RegexExpressionHelper.php [hunk 1]
```diff
@@ -66,4 +69,59 @@ public function resolve(Expr $expr): Type
...
 	}
 
+	public function getPatternModifiers(string $pattern): ?string
+	{
+		$delimiter = $this->getDelimiterFromString(new ConstantStringType($pattern));
+		if ($delimiter === null) {
+			return null;
+		}
+
+		if ($delimiter === '{') {
```
Comment (Seldaek) [inline src/Type/Php/RegexExpressionHelper.php:79]: () and [] are also valid reciprocal delimiters.
<> too but that's a shitty one to use really.. Still probably should support for completeness

https://www.php.net/manual/en/regexp.reference.delimiters.php

**Review by Seldaek — COMMENTED** [review]
Looks good to me aside from some missing delimiter support

**Pull request merged**